src/main/java/algorithms/curated170/medium/LongestSubstringwithAtMostKDistinctCharacters.java

```diff
@@ -0,0 +1,40 @@
+package algorithms.curated170.medium;
+  
+
+public class LongestSubstringwithAtMostKDistinctCharacters {
+
+    public static void main(String[] args) {
+    
+        LongestSubstringwithAtMostKDistinctCharacters lo = new LongestSubstringwithAtMostKDistinctCharacters();
+        int output = lo.lengthOfLongestSubstringKDistinct("abbccc", 2);
+        System.out.println(output);
+    }
+
+    public int lengthOfLongestSubstringKDistinct(String s, int k) {
+       if(k >= s.length()) return s.length();
+      
+        char[] counterMap = new char[256];
```

Comment:
With further consideration, this is actually a char array that is used to hold information about numbers, namely, how many times a give char at index of its ASCII value occurs. Sort of like a "unsigned short".

Reply:
It's sort of unusual, as in normal char arrays: key = int and value = char,
but here: key = char and value = int.